reinforcement_learning/agents/q_learning_agent/q_learning_agent.py

```diff
@@ -44,6 +45,7 @@ def receive_reward(self, reward):
     def exit(self, terminal_state):
         self._update(terminal_state)
         self.performance_measure.append(self._current_episode_return)
+        self.restart()
```

Comment:
Ugly, because you call "interface" method as private method. Better option is to make private method, e.g. _cleaning() and call it at the end of exit() and in restart()

Reply:
True